docs/en/getting-started/get-started-docker.asciidoc

```diff
@@ -1,69 +1,269 @@
+* An enrollment token is generated for {kib}.
+
+You can then {kibana-ref}/docker.html[start {kib}] and enter the enrollment
+token, which is valid for 30 minutes. This token automatically applies the
+security settings from your {es} cluster, authenticates to {es} with the
+`kibana_system` user, and writes the security configuration to `kibana.yml`.
+
+[discrete]
+==== Prerequisites
+. Install the appropriate https://docs.docker.com/get-docker/[Docker application]
+for your operating system.
++
+[NOTE]
+====
+Make sure that Docker is allotted at least 4GiB of memory. In Docker Desktop,
+you configure resource usage on the Advanced tab in Preference (macOS) or
+Settings (Windows).
+====
+
+. Pull the {es} Docker image.
++
+--
+ifeval::["{release-state}"=="unreleased"]
 
-To get the default distributions of {es} and {kib} up and running in Docker,
-you can use Docker Compose.
+WARNING: Version {version} of {es} has not yet been released, so no
+Docker image is currently available for this version.
 
-. Create a `docker-compose.yml` file for the Elastic Stack.
-The following example brings up a three node cluster and Kibana so you can see how things work.
-This all-in-one configuration is a handy way to bring up your first dev cluster before
-you build a distributed deployment with multiple hosts.
+endif::[]
+
+ifeval::["{release-state}"!="unreleased"]
+
+[source,sh,subs="attributes"]
+----
+docker pull docker.elastic.co/elasticsearch/elasticsearch:{version}
+----
+
+endif::[]
+--
+
+. Pull the {kib} Docker image.
 +
+--
 ifeval::["{release-state}"=="unreleased"]
-NOTE: Version {version} of {es} has not been released,
-so the sample compose file is not yet available for this version.
-See the {stack-gs-current}/get-started-docker.html[current version] for the latest sample files.
+
+WARNING: Version {version} of {kib} has not yet been released, so no
+Docker image is currently available for this version.
+
 endif::[]
+
 ifeval::["{release-state}"!="unreleased"]
-["source","yaml",subs="attributes"]
---------------------------------------------
-include::docker/docker-compose.yml[]
---------------------------------------------
+
+[source,sh,subs="attributes"]
+----
+docker pull docker.elastic.co/kibana/kibana:{version}
+----
+
+endif::[]
+--
+
+[[docker-start-enroll]]
+[discrete]
+==== Start a single-node {es} cluster and enroll {kib}
+
+The following command starts a single-node {es} cluster for development or
+testing.
+
+. Start {es} in Docker. A password is generated for the `elastic` user and
+output to the terminal, plus an enrollment token for enrolling {kib}.
++
+--
+ifeval::["{release-state}"=="unreleased"]
+
+WARNING: Version {version} of {es} has not yet been released, so no
+Docker image is currently available for this version.
+
+endif::[]
+
+ifeval::["{release-state}"!="unreleased"]
+[source,sh,subs="attributes"]
+----
+docker run --name es-node01 --net elastic -p 9200:9200 -it docker.elastic.co/elasticsearch/elasticsearch:{version}
```

Comment:
if we don't expose 9300, then no other node can join this cluster. I'm not sure we want to do this, but if we do we should point it out at least.

Reply:
Actually, other nodes running on the same host can join the cluster. I'm not sure if we want to support multi-node from multiple hosts running docker in the getting started experience, as I assume that if you run with docker, you use the same host (but I may be wrong).